Sources/ElevenLabs/Internal/Conversation/Startup/StartupStep.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+
+/// Protocol for individual startup steps in the conversation initialization process.
+/// Each step is responsible for a specific phase of the startup sequence.
+protocol StartupStep {
+    /// The name of this step for logging purposes
+    var stepName: String { get }
+
+    /// Execute the step
+    /// - Throws: Any error that occurs during execution
+    func execute() async throws
+}
+
+/// A startup step that can be retried with configurable delays
+protocol RetryableStartupStep: StartupStep {
+    /// The delays between retry attempts (empty = no retries)
+    var retryDelays: [TimeInterval] { get }
+
+    /// Execute a single attempt of the step
+    /// - Throws: Any error that occurs during execution
+    func executeAttempt() async throws
+}
+
+extension RetryableStartupStep {
+    func execute() async throws {
+        let delays = retryDelays.isEmpty ? [0] : retryDelays
+
+        for (index, delay) in delays.enumerated() {
+            let attemptNumber = index + 1
+
+            if delay > 0 {
+                try? await Task.sleep(nanoseconds: UInt64(delay * 1_000_000_000))
```

Comment:
If the calling code cancels this will just return nil and keep executing. I would remove the `?` so cancellation can propagate up.